owsi-core/owsi-core-components/owsi-core-component-jpa/src/main/java/fr/openwide/core/jpa/batch/executor/BatchRunnable.java

```diff
@@ -0,0 +1,32 @@
+package fr.openwide.core.jpa.batch.executor;
+
+import java.util.List;
+
+public abstract class BatchRunnable<E> {
+
+	public void preExecute(List<Long> allIds) {
+	}
+
+	public void executePartition(List<E> partition) {
+		preExecutePartition(partition);
+
+		for (E entity : partition) {
+			executeUnit(entity);
+		}
+
+		postExecutePartition(partition);
+	}
+
+	public void preExecutePartition(List<E> partition) {
+	}
+
+	public void executeUnit(E unit) {
```

Comment:
This should at least be protected.

On a side note, overriding `executePartition()` currently has the side-effect of making `executeUnit()` useless, which is a bit confusing. Maybe we should provide two different abstract classes in order to avoid this? One would have an abstract executePartition, and the other would extend the first, implement execute partition, and have a abstract executeUnit.